changes to sort columns in a view using keys

jsat97 opened the pull request:
change to sort a view using keys (currently only a mouse click on the column header sorts)

--
Jai

ldegio commented:
Overall, it looks like a really nice and useful feature. Here are some notes:

- I wouldn't use '1' as the key to pop up the menu , since it's too common of a key and it's not consistent with the other popup keys. I instead propose F9 and '>' (the latter is consistent with htop)
- the first time I click on '1', no selection is made on the side menu and the first entry is highlighted. See  for examplehttps://dl.dropboxusercontent.com/u/16649518/Screenshot%202015-12-21%2012.33.26.png: PID is highlighted but CPU is the sorting column
- why highlighting the selection in blue instead of using the already present line highlight feature? Note by the way that the new blue highlighting messes up with the other side panels (see https://dl.dropboxusercontent.com/u/16649518/Screenshot%202015-12-21%2012.39.58.png)
- ESC should close the sorting panel
- it would be nice to have sorting "shortcuts" that don't involve opening the side panel. For example shift+[1-9]
- not crtitical: a visual hint if the sorting is ascending or descending (for example a '^' or a 'v' for each line of the sidepanel) would be great

jsat97 commented:
On Tue, Dec 22, 2015 at 2:13 AM, Loris Degioanni <notifications@github.com>
wrote:

> Overall, it looks like a really nice and useful feature. Here are some
> notes:
>
>    - I wouldn't use '1' as the key to pop up the menu , since it's too
>    common of a key and it's not consistent with the other popup keys. I
>    instead propose F9 and '>' (the latter is consistent with htop)
>
>   will change the sort key to > and F9


>    - the first time I click on '1', no selection is made on the side menu
>    and the first entry is highlighted. See for examplehttps://
>    dl.dropboxusercontent.com/u/16649518/Screenshot%202015-12-21%2012.33.26.png:
>    PID is highlighted but CPU is the sorting column
>
> The highlight was meant to show the current sort column - because its not
visible right away to the user.  I ll check the initial highlight - if the
sort is CPU, it should be highlighted.

>
>    - why highlighting the selection in blue instead of using the already
>    present line highlight feature? Note by the way that the new blue
>    highlighting messes up with the other side panels (see
>    https://dl.dropboxusercontent.com/u/16649518/Screenshot%202015-12-21%2012.39.58.png
>    )
>
> Could you please point out to an example where we use line highlight -
this is not the green box that highlights the current menu selection -
right ? I wanted a persistent highlight that stays as long as the view is
unchanged, and couldn't find any other way to do it.


>    - ESC should close the sorting panel
>
> I had a feeling this would be needed for consistency - so we could have
ESC as well as hitting the sort key (> or F9) again close the sorting panel
?

>
>    - it would be nice to have sorting "shortcuts" that don't involve
>    opening the side panel. For example shift+[1-9]
>
> So we could have shift + <n> to sort colum n  ?



>
>    - not crtitical: a visual hint if the sorting is ascending or
>    descending (for example a '^' or a 'v' for each line of the sidepanel)
>    would be great
>
> Will try this as well

>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-166412118>.
>

ldegio commented:
> Could you please point out to an example where we use line highlight -
> this is not the green box that highlights the current menu selection -
> right ? I wanted a persistent highlight that stays as long as the view is
> unchanged, and couldn't find any other way to do it.

Look at what htop does. It just uses the normal highlight and it makes sure to have the proper element selected when you open the pane. It works pretty well and I would do the same.

> I had a feeling this would be needed for consistency - so we could have
> ESC as well as hitting the sort key (> or F9) again close the sorting panel?

Perfect

> So we could have shift + <n> to sort colum n ?

Yes.

jsat97 commented:
I added some changes based on your suggestions.  The hot keys are hard coded (ugly :-( ) - let me know if there's a better way). Also, let me know if there are other improvements /fixes that need to be done.
--
Jai

jsat97 commented:
Loris

Added changes to #495

--
Jai

On Wed, Dec 23, 2015 at 4:39 AM, Loris Degioanni <notifications@github.com>
wrote:

> Could you please point out to an example where we use line highlight -
> this is not the green box that highlights the current menu selection -
> right ? I wanted a persistent highlight that stays as long as the view is
> unchanged, and couldn't find any other way to do it.
>
> Look at what htop does. It just uses the normal highlight and it makes
> sure to have the proper element selected when you open the pane. It works
> pretty well and I would do the same.
>
> I had a feeling this would be needed for consistency - so we could have
> ESC as well as hitting the sort key (> or F9) again close the sorting
> panel?
>
> Perfect
>
> So we could have shift + to sort colum n ?
>
> Yes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-166758942>.
>

ldegio commented:
looks nice!

Minor nitpick: I see that you added the ability to show the scrolling direction. Would it be possible to improve its look by putting it at the right side of the panel, and maybe putting the direction indicator within brackets, like this (v), (^)

Other than that, I think it's ready to be merged.

ldegio commented:
also, it looks like F9 is not documented in the F1 help screen, and that neiter F9 nor the SHIFT+number shortcuts are mentioned in the manual.

jsat97 commented:
Sure. Adding the braces should be a minor change.
Right side of panel- its already on the right side - any other change  here
?
Jai
On 5 Jan 2016 02:44, "Loris Degioanni" <notifications@github.com> wrote:

> looks nice!
>
> Minor nitpick: I see that you added the ability to show the scrolling
> direction. Would it be possible to improve its look by putting it at the
> right side of the panel, and maybe putting the direction indicator within
> brackets, like this (v), (^)
>
> Other than that, I think it's ready to be merged.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-168810797>.
>

ldegio commented:
Right now, unless I'm missing something, they are just at the right of the text. I was proposing to have them at the far right of the sorting panel.

jsat97 commented:
ok - so that would be a fixed location as opposed to being right adjusted
to the text which could be of varying length. I agree that appending it to
the text was the easiest but not the best thing to do.

--
Jai

On Tue, Jan 5, 2016 at 12:04 PM, Loris Degioanni <notifications@github.com>
wrote:

> Right now, unless I'm missing something, they are just at the right of the
> text. I was proposing to have them at the far right of the sorting panel.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-168916571>.
>

jsat97 commented:
Made the change. please review.
moved sort order indicator to right of side panel
<https://github.com/jsat97/sysdig/commit/b39c85524b1be5d6571fcb43c70cb4870089b974>
b39c855
<https://github.com/jsat97/sysdig/commit/b39c85524b1be5d6571fcb43c70cb4870089b974>

--
Jai


On Tue, Jan 5, 2016 at 12:04 PM, Loris Degioanni <notifications@github.com>
wrote:

> Right now, unless I'm missing something, they are just at the right of the
> text. I was proposing to have them at the far right of the sorting panel.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-168916571>.
>

ldegio commented:
Looks good!

Would you mind fixing the conflicts? At that point I'm ready to merge.

jsat97 commented:
Sure. I think it's a git pull and resolve on my local master followed by
push to github right?
Jai
On 5 Jan 2016 23:18, "Loris Degioanni" <notifications@github.com> wrote:

> Looks good!
>
> Would you mind fixing the conflicts? At that point I'm ready to merge.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-169078340>.
>

jsat97 commented:
After sync with the main master, I see these errors during make  (I have
gcc/g++  4.6.3) --

In file included from /home/jai/sysdig/userspace/libsinsp/k8s.h:11:0,
                 from /home/jai/sysdig/userspace/libsinsp/filterchecks.h:21,
                 from /home/jai/sysdig/userspace/libsinsp/chisel.cpp:36:
/home/jai/sysdig/userspace/libsinsp/k8s_component.h:47:24: sorry,
unimplemented: non-static data member initializers
/home/jai/sysdig/userspace/libsinsp/k8s_component.h:47:24: error: ISO C++
forbids in-class initialization of non-const static member ‘m_port’
/home/jai/sysdig/userspace/libsinsp/k8s_component.h:307:24: sorry,
unimplemented: non-static data member initializers
/home/jai/sysdig/userspace/libsinsp/k8s_component.h:307:24: error: ISO C++
forbids in-class initialization of non-const static member ‘m_port’
/home/jai/sysdig/userspace/libsinsp/k8s_component.h:309:31: sorry,
unimplemented: non-static data member initializers

etc...
make[2]: *** [userspace/libsinsp/CMakeFiles/sinsp.dir/chisel.cpp.o] Error 1
make[1]: *** [userspace/libsinsp/CMakeFiles/sinsp.dir/all] Error 2
make: *** [all] Error 2



On Wed, Jan 6, 2016 at 12:40 AM, Jai <jai_97@rediffmail.com> wrote:

> Sure. I think it's a git pull and resolve on my local master followed by
> push to github right?
> Jai
> On 5 Jan 2016 23:18, "Loris Degioanni" <notifications@github.com> wrote:
>
>> Looks good!
>>
>> Would you mind fixing the conflicts? At that point I'm ready to merge.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/draios/sysdig/pull/495#issuecomment-169078340>.
>>
>

jsat97 commented:
I believe its done. I used gcc 4.7 and it worked. synced with master and
pushed back to my repo. Please merge if all ok.

--
Jai

On Wed, Jan 6, 2016 at 12:32 PM, Jai <jai_97@rediffmail.com> wrote:

> After sync with the main master, I see these errors during make  (I have
> gcc/g++  4.6.3) --
>
> In file included from /home/jai/sysdig/userspace/libsinsp/k8s.h:11:0,
>                  from
> /home/jai/sysdig/userspace/libsinsp/filterchecks.h:21,
>                  from /home/jai/sysdig/userspace/libsinsp/chisel.cpp:36:
> /home/jai/sysdig/userspace/libsinsp/k8s_component.h:47:24: sorry,
> unimplemented: non-static data member initializers
> /home/jai/sysdig/userspace/libsinsp/k8s_component.h:47:24: error: ISO C++
> forbids in-class initialization of non-const static member ‘m_port’
> /home/jai/sysdig/userspace/libsinsp/k8s_component.h:307:24: sorry,
> unimplemented: non-static data member initializers
> /home/jai/sysdig/userspace/libsinsp/k8s_component.h:307:24: error: ISO C++
> forbids in-class initialization of non-const static member ‘m_port’
> /home/jai/sysdig/userspace/libsinsp/k8s_component.h:309:31: sorry,
> unimplemented: non-static data member initializers
>
> etc...
> make[2]: *** [userspace/libsinsp/CMakeFiles/sinsp.dir/chisel.cpp.o] Error 1
> make[1]: *** [userspace/libsinsp/CMakeFiles/sinsp.dir/all] Error 2
> make: *** [all] Error 2
>
>
>
> On Wed, Jan 6, 2016 at 12:40 AM, Jai <jai_97@rediffmail.com> wrote:
>
>> Sure. I think it's a git pull and resolve on my local master followed by
>> push to github right?
>> Jai
>> On 5 Jan 2016 23:18, "Loris Degioanni" <notifications@github.com> wrote:
>>
>>> Looks good!
>>>
>>> Would you mind fixing the conflicts? At that point I'm ready to merge.
>>>
>>> —
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/draios/sysdig/pull/495#issuecomment-169078340>.
>>>
>>
>

ldegio commented:
One more thing: it looks like you are using spaces instead of tabs for formatting, which conflicts with the rest of the sysdig code base and makes the patch look ugly. 

Could you please revert to tabs?

jsat97 commented:
Ok. Will do. It doesn't show up in vim, its in my vim setting.
On 6 Jan 2016 23:18, "Loris Degioanni" <notifications@github.com> wrote:

> One more thing: it looks like you are using spaces instead of tabs for
> formatting, which conflicts with the rest of the sysdig code base and makes
> the patch look ugly.
>
> Could you please revert to tabs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-169401804>.
>

jsat97 commented:
Could you please check now. I converted the spaces to tabs and also cleaned
up.
I made a clean build on my VM and that was ok - however, I see that travis
build failed. Couldn't tell much from the logs.
--


On Thu, Jan 7, 2016 at 7:34 AM, Jai <jai_97@rediffmail.com> wrote:

> Ok. Will do. It doesn't show up in vim, its in my vim setting.
> On 6 Jan 2016 23:18, "Loris Degioanni" <notifications@github.com> wrote:
>
>> One more thing: it looks like you are using spaces instead of tabs for
>> formatting, which conflicts with the rest of the sysdig code base and makes
>> the patch look ugly.
>>
>> Could you please revert to tabs?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/draios/sysdig/pull/495#issuecomment-169401804>.
>>
>

jsat97 commented:
just noticed your comment about documenting F9  and shift<1-9> in the help
menu.
the latter was already documented under keyboard shortcuts for the view
window, but for F9,  I was nt sure if that was the right place as it is not
a short cut. I could put it there - please confirm
Also, do we want to put > along with F9, like -
> | F9
--
Jai

On Thu, Jan 7, 2016 at 5:10 PM, Jai <jai_97@rediffmail.com> wrote:

> Could you please check now. I converted the spaces to tabs and also
> cleaned up.
> I made a clean build on my VM and that was ok - however, I see that travis
> build failed. Couldn't tell much from the logs.
> --
>
>
> On Thu, Jan 7, 2016 at 7:34 AM, Jai <jai_97@rediffmail.com> wrote:
>
>> Ok. Will do. It doesn't show up in vim, its in my vim setting.
>> On 6 Jan 2016 23:18, "Loris Degioanni" <notifications@github.com> wrote:
>>
>>> One more thing: it looks like you are using spaces instead of tabs for
>>> formatting, which conflicts with the rest of the sysdig code base and makes
>>> the patch look ugly.
>>>
>>> Could you please revert to tabs?
>>>
>>> —
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/draios/sysdig/pull/495#issuecomment-169401804>.
>>>
>>
>

jsat97 commented:
added F9 to help.

--


On Fri, Jan 8, 2016 at 11:49 AM, Jai <jai_97@rediffmail.com> wrote:

> just noticed your comment about documenting F9  and shift<1-9> in the help
> menu.
> the latter was already documented under keyboard shortcuts for the view
> window, but for F9,  I was nt sure if that was the right place as it is not
> a short cut. I could put it there - please confirm
> Also, do we want to put > along with F9, like -
> > | F9
> --
> Jai
>
> On Thu, Jan 7, 2016 at 5:10 PM, Jai <jai_97@rediffmail.com> wrote:
>
>> Could you please check now. I converted the spaces to tabs and also
>> cleaned up.
>> I made a clean build on my VM and that was ok - however, I see that
>> travis build failed. Couldn't tell much from the logs.
>> --
>>
>>
>> On Thu, Jan 7, 2016 at 7:34 AM, Jai <jai_97@rediffmail.com> wrote:
>>
>>> Ok. Will do. It doesn't show up in vim, its in my vim setting.
>>> On 6 Jan 2016 23:18, "Loris Degioanni" <notifications@github.com> wrote:
>>>
>>>> One more thing: it looks like you are using spaces instead of tabs for
>>>> formatting, which conflicts with the rest of the sysdig code base and makes
>>>> the patch look ugly.
>>>>
>>>> Could you please revert to tabs?
>>>>
>>>> —
>>>> Reply to this email directly or view it on GitHub
>>>> <https://github.com/draios/sysdig/pull/495#issuecomment-169401804>.
>>>>
>>>
>>
>

ldegio commented:
hmm, I'm probably missing something, but I don't see it documented in the .md page in userspace/sysdig/man?

jsat97 commented:
sorry. missed the man page (I only changed the help menu in the code
before). fixed now in csysdig.md

--


On Sat, Jan 9, 2016 at 2:01 AM, Loris Degioanni <notifications@github.com>
wrote:

> hmm, I'm probably missing something, but I don't see it documented in the
> .md page in userspace/sysdig/man?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-170115148>.
>

jsat97 commented:
sorry. missed the man page (I only changed the help menu in the code
before). fixed now in csysdig.md

--

On Sat, Jan 9, 2016 at 2:01 AM, Loris Degioanni <notifications@github.com>
wrote:

> hmm, I'm probably missing something, but I don't see it documented in the
> .md page in userspace/sysdig/man?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/draios/sysdig/pull/495#issuecomment-170115148>.
>

ldegio commented:
Merged. Thanks!
